### ISSUE TITLE
Auditing should report the shard copy ID that failed and patching should remove it from the manifest

### ISSUE THREAD

**Issue by WilfredTA** [layr-team/Layr#44]
The only case in which an audit fails but the shard copy ID is still accessible is when the host node was offline at the time of the audit. In all other cases, it is in the data owner's best interest to completely forget about that shard copy ID forever.

Therefore, auditing should report the copy ID that failed and patching should remove that copy ID from the manifest.

In the future, we can always add three states to an audit rather than two states (true or false). The third state will handle the case in which the audited node is simply offline. That's a relatively easy thing to do once the above change is made, since the node alive test is a simple ping with an event handler: if the ping fails, set the result of the shard copy ID to this third state.

**Comment by dylankb** [comment]
Yes, we should definitely do this! I'd add two small things. One, we should couple this change with creating a system to alert hosts that their modified shard has now become a "zombie". A log file with the original filename and SHA of the original shard that will hopefully be enough to alert them that they are free to delete the hosted file. This isn't important for bad actors, but helps good ones keep their file systems clean.

Two, I think there's a fourth possible state for a file - deleted. We should make sure to handle that as well.

**Comment by WilfredTA** [comment]
For the first case, if a host modified a file, wouldn't you consider them a bad actor?

Also, the second case is already handled.

**Comment by dylankb** [comment]
I think it's certainly possible that someone could accidentally edit a shard without meaning to. For example, if you open a shard file in the Atom text editor and save the file, Atom will automatically insert a new line character by default. This is a pretty subtle change, but would make the shard fail the audit. But what do you think?

Oh good!

**Comment by WilfredTA** [comment]
That's a fair point, yeah. I believe that if the newline was deleted, the shard would still fail the audit, but I'm not sure. I tried experimenting with adding and deleting content of a shard and the audit failed even after the added content was removed. It's worth experimenting and researching that.

Yeah, it does seem pretty easy to accidentally manipulate a file, then. We can certainly implement a notification system to take care of zombie shards, then.

So, if an audit fails we can let the failing host node know (if they're online), and if they're not online, well the result of the shard audit will be that 3rd state, which is handled by generating a new copy which is then distributed, but keeping it in the manifest file
